components/elective/ElectiveStudentListCard.tsx

```diff
@@ -0,0 +1,120 @@
+import EnrollmentIndicator from "@/components/elective/EnrollmentIndicator";
+import PersonCard from "@/components/person/PersonCard";
+import cn from "@/utils/helpers/cn";
+import getLocaleName from "@/utils/helpers/getLocaleName";
+import useLocale from "@/utils/helpers/useLocale";
+import { LangCode, StylableFC } from "@/utils/types/common";
+import { ElectiveSubject } from "@/utils/types/elective";
+import { UserRole } from "@/utils/types/person";
+import {
+  AssistChip,
+  ChipSet,
+  DURATION,
+  EASING,
+  MaterialIcon,
+  Search,
+  Text,
+  transition,
+} from "@suankularb-components/react";
+import { motion } from "framer-motion";
+import { useTranslation } from "next-i18next";
+import Link from "next/link";
+import { useRouter } from "next/router";
+import { sort } from "radash";
+import { useState } from "react";
+
+/**
+ * A Card that displays a list of all Students enrolled in an Elective Subject,
+ * filterrable by name.
+ *
+ * @param electiveSubject The Elective Subject to display the Students of.
+ */
+const ElectiveStudentListCard: StylableFC<{
+  electiveSubject: ElectiveSubject;
+}> = ({ electiveSubject, style, className }) => {
+  const locale = useLocale();
+  const { locales } = useRouter();
+  const { t } = useTranslation("elective", { keyPrefix: "detail.students" });
+
+  const [query, setQuery] = useState("");
+
+  const classrooms = Object.fromEntries(
+    electiveSubject.applicable_classrooms.map((classroom) => [
+      classroom.id,
+      classroom,
+    ]),
+  );
+  const students = sort(
+    sort(
+      electiveSubject.students.filter((student) =>
+        locales?.some((locale) =>
+          getLocaleName(locale as LangCode, student).includes(query),
+        ),
+      ),
+      (student) => student.classroom?.number || 1000,
```

Comment:
### About `sort`

Think of the `sort` function as mapping and then sorting the array. This is an array of Students, we’re converting it into an array of 3-digit classroom numbers and then sorting that array.

That array would look like this:

```ts
[302, undefined, 301, undefined, undefined]
```

### About `|| 1000`

However, some Students aren’t in a Classroom, and `student.classroom?.number` would return `undefined`, which we can’t use to sort. To solve this, we use short-circuit evaluation, where it would use `student.classroom?.number` if available, and default to `1000` if not.

```ts
student.classroom?.number || 1000
undefined || 1000
1000
```

```ts
[302, 1000, 301, 1000, 1000]
```

Much better! We use 1000 because the maximum Classroom number theoretically is M.999, thus Students without a Classroom would be sorted to the end of the list.

```ts
[301, 302, 1000, 1000, 1000]
```